examples/cmd/vppagent-icmp-responder-nse/nse.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"sync"
+	"syscall"
+
+	"github.com/ligato/networkservicemesh/controlplane/pkg/nsmd"
+
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/registry"
+
+	"github.com/ligato/networkservicemesh/pkg/tools"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+const (
+	// networkServiceName defines Network Service Name the NSE is serving for
+	NetworkServiceName      = "vppagent-icmp-responder"
+	DefaultVPPAgentEndpoint = "localhost:9112"
+)
+
+func main() {
+	var wg sync.WaitGroup
+
+	nsmServerSocket, _ := os.LookupEnv(nsmd.NsmServerSocketEnv)
+	logrus.Infof("nsmServerSocket: %s", nsmServerSocket)
+	// TODO handle missing env
+
+	nsmClientSocket, _ := os.LookupEnv(nsmd.NsmClientSocketEnv)
+	logrus.Infof("nsmClientSocket: %s", nsmClientSocket)
+	// TODO handle missing env
+
+	workspace, _ := os.LookupEnv(nsmd.WorkspaceEnv)
```

Comment:
Now this env is set by DevicePluginAPI https://github.com/ligato/networkservicemesh/pull/513/files#diff-63d6957eaddf3c7e04797de53e05457cR66. Or you mean that DevicePluginAPI should return it as a reply?

Reply:
I mean it should be returned as an Env variable [in the nsmdp](https://github.com/ligato/networkservicemesh/blob/master/k8s/cmd/nsmdp/nsmdp.go#L63) which will cause it to be present in the Pod.

As to where the value comes from in nsmdp, it should be added to [ClientConnectionReply](https://github.com/ligato/networkservicemesh/blob/master/controlplane/pkg/apis/nsmdapi/nsmd.proto#L13) which is [returned by the nsmd to the nsmdp](https://github.com/ligato/networkservicemesh/blob/master/controlplane/pkg/nsmd/nsmd.go#L75)

This way, rather than cross linking a bunch of constants, that env variable starts in nsmd, gets returned in ClientConnectionReply to the nsmdp, the nsmdp puts it in the AllocateResponse it returns the kubelet, and kubelet will inject the Env variable into the Pod :)